### PR TITLE
Make lb-method & lb-provider configurable for openstack provider

### LINKED ISSUE #13532: Make additional OpenStack LoadBalancer cloud-config parameters for user clusters configurable per datacenter
### Use Case & Description of the feature you would like to add

A cloud-config is automatically generated for every user cluster. Some of the parameters come from the cluster resource (e.g. username and password), and a few can be specified e.g. in the DataCenter definition. However, some are unfortunately hardcoded.

In our new environment, the target setup uses OpenStack Octavia's ovn instead of amphora as the LB provider. Unfortunately, this requires the `lb-provider` as well as the `lb-method` to be configured to different values than what is currently hardcoded in KKP.

Thus our request is to make these two parameters configurable. Exposing them through the DatacenterSpec, as some of the other cloud-config parameters in the same group of options, would be sufficient for our current use case.

### Solution details

- if desired, the KKP admin can specify the `lb-provider` and `lb-method` parameters on a per-datacenter level
- this would be similar to how `manage-security-groups` can already be configured: https://docs.kubermatic.com/kubermatic/v2.25/references/crds/#datacenterspecopenstack
-

### Alternative approaches

Only let KKP generate the `[Global]` block of the cloud-config and implement a single multiline parameter that is simply added verbatim after the `[Global]` block instead, so we have full control over the rest of the cloud-config (which would include the existing `[LoadBalancer]` and `[BlockStorage]` blocks, but the OS CCM actually knows even more options, e.g. `[Networking]`, as well. Of course the current output should probably still be the default, if not overwritten.

### Additional information

This is what the `LoadBalancer` section of a UC cloud-config currently looks like:
```
[LoadBalancer]
lb-version = "v2"
subnet-id = ""
floating-network-id = ""
lb-method = "ROUND_ROBIN"
lb-provider = ""
manage-security-groups = true # configurable
```

... which works well enough for amphora-backed LBs (although some people might want to change the `lb-method` even then).

This is what I need it to look like - note that ovn-backed LBs only support this one `lb-method` (which is in turn not supported by amphora):
```
[LoadBalancer]
lb-version = "v2"
subnet-id = ""
floating-network-id = ""
lb-method = "SOURCE_IP_PORT" # configurable
lb-provider = "ovn" # configurable
manage-security-groups = true # configurable
```

We've already tested this config because in KubeOne we can specify the cloud-config freely.

Edit: I also paused a user cluster, edited its cloud-config, restarted its OS CCM pod and then load balancer creation worked with ovn.

### PR DESCRIPTION
**Pull request by akash-gautam**
**What this PR does / why we need it**:
Allow KKP admins to configure "lb-method" & "lb-provider" at the datacenter for openstack provider.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubermatic/kubermatic/issues/13532

**What type of PR is this?**
Add one of the following kinds:
/kind feature
/kind api-change

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Loadbalancer provider (lb-provider) & loadbalancer method (lb-method) can be configured at the datacenter for openstack provider.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/resources/cloudconfig/cloudconfig.go [hunk 1]
```diff
@@ -100,6 +100,14 @@ func CloudConfig(
 
 	case cloud.Openstack != nil:
 		manageSecurityGroups := dc.Spec.Openstack.ManageSecurityGroups
+		loadBalancerProvider := ""
+		loadBalancerMethod := "ROUND_ROBIN"
```
Comment (ahmedwaleedmalik) [inline pkg/resources/cloudconfig/cloudconfig.go]: Please create a constant for this at the package/file level to make it more visible.

**Comment by ahmedwaleedmalik** [comment]
/cc @ahmedwaleedmalik 
One nit, rest looks good but will take another look later

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve
/hold 
CentOS tests are not supposed to work anymore, please either remove them in your PR or ask @xrstf for overrides. 

Only put the `hold` to avoid retries, please feel free to remove it when required.

**Comment by xrstf** [comment]
/lgtm cancel

New golangci-lint doesn't like the complexity :grin:

**Pull request merged**